packages/ui-extensions-server-kit/src/ExtensionServerClient/ExtensionServerClient.ts

```diff
@@ -197,16 +204,45 @@ export class ExtensionServerClient implements ExtensionServer.Client {
         ...extension,
         localization,
         extensionPoints: isUIExtension(extension)
-          ? extension.extensionPoints?.map((extensionPoint) => ({
-              ...extensionPoint,
-              localization,
-            }))
+          ? this._getLocalizedExtensionPoints(shouldUpdateTranslations, localization, extension.extensionPoints)
           : extension.extensionPoints,
       }
 
       return this.extensionsByUuid[extension.uuid]
     })
   }
+
+  private _getLocalizedExtensionPoints(
+    shouldUpdateTranslations: boolean,
+    localization: FlattenedLocalization | Localization | null | undefined,
+    extensionPoints: ExtensionPoint[],
+  ): ExtensionPoint[] {
+    return extensionPoints?.map((extensionPoint) => {
+      if (
+        shouldUpdateTranslations &&
+        localization &&
+        isFlattenedTranslations(localization) &&
+        extensionPoint.label &&
+        extensionPoint.label.startsWith('t:')
+      ) {
+        return {
+          ...extensionPoint,
+          localization,
+          label: this._getLocalizedLabel(localization, extensionPoint.label),
```

Comment:
Can you move this outside the loop - parsing JSON is pretty slow so it's better if we only do it once per extension